File: clients/javascript/model/unit-action.js
const Vec2Double = require('./vec2-double').Vec2Double;
class UnitAction {
    constructor(velocity, jump, jumpDown, aim, shoot, reload, swapWeapon, plantMine) {
        this.velocity = velocity;
        this.jump = jump;
        this.jumpDown = jumpDown;
        this.aim = aim;
        this.shoot = shoot;
        this.reload = reload;
        this.swapWeapon = swapWeapon;
        this.plantMine = plantMine;
    }
    static async readFrom(stream) {
        let velocity;
        velocity = await stream.readDouble();
        let jump;
        jump = await stream.readBool();
        let jumpDown;
        jumpDown = await stream.readBool();
        let aim;
        aim = await Vec2Double.readFrom(stream);
        let shoot;
        shoot = await stream.readBool();
        let reload;
        reload = await stream.readBool();
        let swapWeapon;
        swapWeapon = await stream.readBool();
        let plantMine;
        plantMine = await stream.readBool();
        return new UnitAction(velocity, jump, jumpDown, aim, shoot, reload, swapWeapon, plantMine);
    }
    async writeTo(stream) {
        let velocity = this.velocity;
        await stream.writeDouble(velocity);
        let jump = this.jump;
        await stream.writeBool(jump);
        let jumpDown = this.jumpDown;
        await stream.writeBool(jumpDown);
        let aim = this.aim;
        await aim.writeTo(stream);
        let shoot = this.shoot;
        await stream.writeBool(shoot);
        let reload = this.reload;
        await stream.writeBool(reload);
        let swapWeapon = this.swapWeapon;
        await stream.writeBool(swapWeapon);
        let plantMine = this.plantMine;
        await stream.writeBool(plantMine);
    }
}
module.exports = { UnitAction: UnitAction }

File: clients/python/model/unit_action.py
from .vec2_double import Vec2Double
class UnitAction:
    def __init__(self, velocity, jump, jump_down, aim, shoot, reload, swap_weapon, plant_mine):
        self.velocity = velocity
        self.jump = jump
        self.jump_down = jump_down
        self.aim = aim
        self.shoot = shoot
        self.reload = reload
        self.swap_weapon = swap_weapon
        self.plant_mine = plant_mine
    @staticmethod
    def read_from(stream):
        velocity = stream.read_double()
        jump = stream.read_bool()
        jump_down = stream.read_bool()
        aim = Vec2Double.read_from(stream)
        shoot = stream.read_bool()
        reload = stream.read_bool()
        swap_weapon = stream.read_bool()
        plant_mine = stream.read_bool()
        return UnitAction(velocity, jump, jump_down, aim, shoot, reload, swap_weapon, plant_mine)
    def write_to(self, stream):
        stream.write_double(self.velocity)
        stream.write_bool(self.jump)
        stream.write_bool(self.jump_down)
        self.aim.write_to(stream)
        stream.write_bool(self.shoot)
        stream.write_bool(self.reload)
        stream.write_bool(self.swap_weapon)
        stream.write_bool(self.plant_mine)
    def __repr__(self):
        return "UnitAction(" + \
            repr(self.velocity) + "," + \
            repr(self.jump) + "," + \
            repr(self.jump_down) + "," + \
            repr(self.aim) + "," + \
            repr(self.shoot) + "," + \
            repr(self.reload) + "," + \
            repr(self.swap_weapon) + "," + \
            repr(self.plant_mine) + \
            ")"
